### ISSUE TITLE
VAT field can't be disabled in customer registration form

### ISSUE THREAD

**Issue by johnulecan** [magento/magento2#35239]
### Preconditions (*)

1. 2.4.x
2. 2.3.x

### Steps to reproduce (*)

1. Go to Admin->Store->Configuration->Customer and set Show Tax/VAT Number either Optional or Required in the Name and Address Options section 
2. Set Show VAT Number on Storefront to No in Admin->Store->Configuration->Customer Create New Account Options section

### Expected result (*)

1. VAT field shouldn't be displayed in the customer registration form in front-end.

### Actual result (*)

1. VAT field is still visible in the customer registration field.

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [x] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by twosg** [comment]
Hi, `taxvat` and `vat_id` are different fields. `taxvat` belongs to the customer entity, `vat_id` is for the customer address entity. The settings should not depend on each other and work correctly as intended.

The setting `Show VAT Number on Storefront` won't have any effect on the registration page until you enable the address fields to show on that page.

**Comment by johnulecan** [comment]
> Hi, `taxvat` and `vat_id` are different fields. `taxvat` belongs to the customer entity, `vat_id` is for the customer address entity. The settings should not depend on each other and work correctly as intended.
> 
> The setting `Show VAT Number on Storefront` won't have any effect on the registration page until you enable the address fields to show on that page.

Hey @twosg Thank you for the feedback. It makes sense for customer address forms, but, on registration form you should be able to remove the taxvat field without editing register.phtml. `Show Tax/VAT Number` settings are located in the `Address Options` section. In the `Create New Account Options` section we should be able to set `taxvat` visibility on registration form. At this point, if you enable  `Show Tax/VAT Number` for address forms, `taxvat` field will be rendered in the registration form also (for a B2B + B2C store you don't want to ask for taxvat in the customer registration form, but you rather want to provide customers the possibility to add both a business and a personal billing/shipping address).

**Comment by twosg** [comment]
I am not against improvements on this but:

You are able to hide the field in the `register.phtml` by setting `Name and Address Options -> Show Tax/VAT Number` to  `No`. This will generally hide the field, if you want to hide it only for the register.phtml then you need to touch the template. I agree that a bit more fine-tunable settings would be preferrable, but the setting at `Create new account options -> Show VAT Number on Storefront` is there to control another field called `vat_id`. It has nothing to do with the `taxvat` field and therefore should not control the visibility of it.

The main source of confusion about these fields is that back in the Magento1 days (until version 1.7) there only was the `taxvat` field and `vat_id` did not exist. Nowadays the field label should be renamed to `Tax Number` instead of `Tax/VAT Number`, because we have an own field for the VAT Number on the customer address with `vat_id`.

**Comment by justpeeet** [comment]
> I am not against improvements on this but:
> 
> You are able to hide the field in the `register.phtml` by setting `Name and Address Options -> Show Tax/VAT Number` to `No`. This will generally hide the field, if you want to hide it only for the register.phtml then you need to touch the template. I agree that a bit more fine-tunable settings would be preferrable, but the setting at `Create new account options -> Show VAT Number on Storefront` is there to control another field called `vat_id`. It has nothing to do with the `taxvat` field and therefore should not control the visibility of it.
> 
> The main source of confusion about these fields is that back in the Magento1 days (until version 1.7) there only was the `taxvat` field and `vat_id` did not exist. Nowadays the field label should be renamed to `Tax Number` instead of `Tax/VAT Number`, because we have an own field for the VAT Number on the customer address with `vat_id`.

That's not right. If you are using Magento 2 as headless system I am relying on the address settings. So If I am not able to disable this I will get an additional field for the customer address which shouldn't be there cause the setting is not working correctly. Changing the template won't help at this point cause the possibility to disable the field in the customer address settings should work as intended

**Comment by engcom-Delta** [comment]
Hi @johnulecan ,
We tried to reproduce the issue on Magento 2.4 Develop & 2.4.3p1 instance. As per provided steps issue is reproducible in below test scenarios.

Scenario1: 
1. Set 'Optional' to  "Show Tax/VAT Number" field in Customer name & address options.
2. Set 'NO' to field "Show VAT Number on Storefront" in Create new Account options
3. Save 
4. Go to frontend and click on customer registration 
5. verify the VAT Number field

Actual:
Tax or Vat Number field displayed on customer registration form.
<img width="1440" alt="Screenshot 2022-04-01 at 9 09 56 AM" src="https://user-images.githubusercontent.com/51681379/161193133-f52507a0-c5e2-4431-a80c-9804517c7523.png">
<img width="1440" alt="Screenshot 2022-04-01 at 9 13 28 AM" src="https://user-images.githubusercontent.com/51681379/161193156-2badc48d-53a7-42ec-966b-72600b8056f8.png">
<img width="1440" alt="Screenshot 2022-04-01 at 9 15 08 AM" src="https://user-images.githubusercontent.com/51681379/161193161-114b7b4f-d76e-4459-a694-5560b8ff0f6b.png">

Scenario2: 
1. Set 'Required' to  "Show Tax/VAT Number" field in Customer name & address options.
2. Set 'NO' to field "Show VAT Number on Storefront" in Create new Account options
3. Save 
4. Go to frontend and click on customer registration 
5. verify the VAT Number field

Actual:
Tax or Vat Number field displayed on customer registration form.

<img width="1440" alt="Screenshot 2022-04-01 at 8 57 31 AM" src="https://user-images.githubusercontent.com/51681379/161192972-e3fb4c5b-e4a8-464b-a3ce-5064ea7c322e.png">
<img width="1440" alt="Screenshot 2022-04-01 at 8 59 55 AM" src="https://user-images.githubusercontent.com/51681379/161193001-087779f6-e5bf-4ed5-a1ea-9456f13eda95.png">
<img width="1440" alt="Screenshot 2022-04-01 at 9 01 57 AM" src="https://user-images.githubusercontent.com/51681379/161193011-08546d7d-b3cd-487c-8096-9f13ff5ff6fe.png">



Scenario3: 
1. Set 'NO' to  "Show Tax/VAT Number" field in Customer name & address options.
2. Set 'NO' to field "Show VAT Number on Storefront" in Create new Account options
3. Save 
4. Go to frontend and click on customer registration 
5. verify the VAT Number field


Actual:
Tax or Vat Number field is not displayed on customer registration form.
<img width="1440" alt="Screenshot 2022-04-01 at 9 06 09 AM" src="https://user-images.githubusercontent.com/51681379/161192837-3dc84a66-14f7-4702-a709-fb4623f36538.png">


As per first two test scenarios's issue is reproducible. Hence confirming the it.

**Comment by engcom-Bravo** [comment]
Hi @johnulecan,

Thanks for your reporting and collaboration.

We have reverified the issue in Latest 2.4-develop instance and the issue is still reproducible.Kindly refer the screenshots.


![Create-New-Customer-Account-11-18-2024_10_15_AM](https://github.com/user-attachments/assets/1fbdc8b9-75d0-426b-bb00-c2d83b7afc4d)

VAT field is still visible in the customer registration field.


Hence Confirming the issue.

Thanks.
